fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/TagUtil.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tag.convention.v1;
+
+import java.util.Optional;
+
+import net.minecraft.tag.TagKey;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryEntry;
+import net.minecraft.util.registry.RegistryKey;
+
+/**
+ * A Helper class for dealing with {@link TagKey}s when their type has no easy way of querying if they are in a tag.
+ */
+public class TagUtil {
+	/**
+	 * @return if the entry is in the provided tag.
+	 */
+	@SuppressWarnings("unchecked")
+	public static <T> boolean isIn(TagKey<T> tagKey, T entry) {
+		Optional<? extends Registry<?>> maybeRegistry = Registry.REGISTRIES.getOrEmpty(tagKey.registry().getValue());
+
+		if (maybeRegistry.isPresent()) {
+			if (tagKey.isOf(maybeRegistry.get().getKey())) {
+				Registry<T> registry = (Registry<T>) maybeRegistry.get();
+
+				Optional<RegistryKey<T>> maybeKey = registry.getKey(entry);
+
+				// Check synced tag
+				return maybeKey.filter(registryKey -> registry.entryOf(registryKey).isIn(tagKey))
+						.isPresent();
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * @return if the entry is in the provided tag.
+	 */
+	public static <T> boolean isIn(TagKey<T> tagKey, RegistryEntry<T> registryEntry) {
```

Comment:
Why? :D

Reply:
Completeness :P
It can be removed if needed.